src/main/java/com/vaadin/recipes/recipe/blockingdialog/BlockingDialog.java

```diff
@@ -0,0 +1,104 @@
+import java.util.function.Consumer;
+
+@Route("blocking-dialog")
+@Metadata(
+        howdoI = "Simulate a blocking dialog as is known from Swing programming or Eclipse RCP",
+        description = "From classic Java programming, often a dialog is shown blocking the rest of the program execution until a user decision is made. In Vaadin, as you cannot block the application server's http response thread, this has to be simulated blocking the UI, and letting another thread execute the continuation code.",
+        tags = { Tag.JAVA }
+)
+
+public class BlockingDialog extends Recipe {
+
+    // This class can be copied into a utility class and is an inner class only because that way the code is
+    // shown in the Vaadin Cookbook environment.
+    public static class YesNoDecider {
+        abstract static class MyCommand implements Command {
+            volatile boolean done;
+        }
+
+        // to be called from within a UI's thread
+        public static void yesNo(String question, Consumer<Boolean> consumer) {
+
+            Dialog dialog = new Dialog();
+
+            CompletableFuture<Boolean> future = new CompletableFuture<>();
+            dialog.add(new Span(question));
+            dialog.add(new Button("yes", e -> future.complete(true)));
+            dialog.add(new Button("no", e -> future.complete(false)));
+            dialog.setModal(true); // in fact, the extra indirection through a command that is re-executed a number of times would allow for non-modality, too.
+            dialog.setCloseOnEsc(false); // allowing to remove the yes/np buttons would keep the user-code un-executed forever, leaking memory and possibly other resources.
+            dialog.setCloseOnOutsideClick(false); // allowing to remove the yes/no buttons would keep the user-code un-executed forever, leaking memory and possibly other resources.
+            dialog.open();
+
+            // Command will be run in the UI thread eventually
+            final YesNoDecider.MyCommand command = new YesNoDecider.MyCommand() {
+                @Override
+                public void execute() {
+                    if (future.isDone()) {
+                        try {
+                            consumer.accept(future.get());
+                            done = true;
+                        } catch (InterruptedException | ExecutionException ex) {
+                            ex.printStackTrace();
+                        }
+                    }
+                }
+            };
+
+            UI ui = UI.getCurrent(); // in the UI thread, get the UI instance
+            new Thread(() -> { // worker thread trying to let the UI thread run the command which in turn runs the user code as soon as the required user-input boolean is there
+                UI.setCurrent(ui); // in case the command uses 'UI.getCurrent()' assign the parent thread's UI to this new child thread.
+                while (!command.done) { // repeat until it is finally done.
```

Comment:
Isn't this a spinlock, i.e. eating 100% CPU while waiting for progress? 

`ui.access(command).get()` will block until the command has had a chance to run, which happens basically immediately on the same thread while there are no other threads locking the same UI. This means that it will immediately get back to running the same line of code again without anything to let it fall back.

If that's the case, then I don't think we should show this as an example since it's not even remotely scalable. If that is not the case, then it would be good to describe why that isn't the case.

Reply:
Yes it is. I had my doubts last night which is why I requested the code review. Originally I was thinking 'access' would inject some code at the and of a worker thread's duty, before the HTTP response would be generated.

If instead it is nothing but a 'synchronized' way of executing code, then I don't understand why there is a a task queue.

In any case, the code needs to be refactored and I am working on it.

Reply:
@Legioth I repaired the stanza, please see new code.